### DISCUSSION TITLE
How RAW file generate?

### DISCUSSION THREAD

**Discussion by tonystarkiss** [deepmodeling/deepmd-kit#904] [Q&A]
I have got a OUTCAR file by VASP , How can I convert it to energy.raw and force.raw?

**Comment by njzjz** [comment]
```py
import dpdata
d_outcar = dpdata.LabeledSystem('OUTCAR')
d_outcar.to('deepmd/raw', 'dpmd_raw')
```
See https://github.com/deepmodeling/dpdata#load-data for detail.

**Reply by njzjz** [reply]
By the way, you can directly generate numpy files using
```py
d_outcar.to('deepmd/npy', 'dpmd_npy')
```

**Reply by tonystarkiss** [reply]
>>> d_poscar = dpdata.System('my.POSCAR')
>>> d_poscar.to("deepmd/raw", "dpmd_raw1")
>>> dpdata.LabeledSystem('OUTCAR')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/dpdata/dpdata/system.py", line 877, in __init__
    self.from_fmt(file_name, fmt, type_map=type_map, begin= begin, step=step, **kwargs)
  File "/data/dpdata/dpdata/system.py", line 118, in from_fmt
    return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)
  File "/data/dpdata/dpdata/system.py", line 884, in from_fmt_obj
    data = fmtobj.from_labeled_system(file_name, **kwargs)
  File "/data/dpdata/dpdata/plugins/vasp.py", line 65, in from_labeled_system
    = dpdata.vasp.outcar.get_frames(file_name, begin=begin, step=step)
  File "/data/dpdata/dpdata/vasp/outcar.py", line 65, in get_frames
    coord, cell, energy, force, virial, is_converge = analyze_block(blk, ntot, nelm)
  File "/data/dpdata/dpdata/vasp/outcar.py", line 128, in analyze_block
    tmp_l = lines[jj]
**IndexError: list index out of range**

**Reply by tonystarkiss** [reply]
How to solve index out of range?

**Reply by njzjz** [reply]
@AnguseZhang

**Comment by amcadmus** [comment]
Could you please provide the OUTCAR so we can reproduce the error.

**Reply by tonystarkiss** [reply]
I have solved the issue, thanks.
my OUTCAR file the last Iteration loop is abnormal interrupted, when I delete the uncompleted loop, dpdata.LabeledSystem('OUTCAR') succeed.
